src/components/page-donate/page-donate.tsx

```diff
@@ -19,54 +8,50 @@ export class PageDonate {
   @State() private amount?: number | null;
   @State() private showConfirmation: boolean = false;
   @State() private canNativeShare: boolean = false;
+  @State() private referral?: string | null;
 
   public componentWillLoad() {
     document.title = `Donate | Pizza to the Polls`;
-  }
+    const urlParams = new URLSearchParams(window.location.search);
+    this.referral = urlParams.get("referral") || "";
 
-  public render() {
-    let handler: any = null;
-    const StripeCheckout: any = (window as any).StripeCheckout;
+    if (urlParams.get("success") && urlParams.get("amount_usd")) {
+      this.amount = parseInt(urlParams.get("amount_usd") as string, 10);
+      this.showConfirmation = true;
+    }
+  }
 
-    // Get referral from URL
-    const urlParams = new URLSearchParams(window.location.search);
-    const referral = urlParams.get("referral") || "";
+  public async donate(amount: number) {
+    const resp = await fetch(`${process.env.PIZZA_BASE_DOMAIN}/donations`, {
+      body: JSON.stringify({ amountUsd: amount, referrer: this.referral }),
+      method: "POST",
+      mode: "cors",
+      headers: { "Content-Type": "application/json" },
+    });
+
+    if (resp.status === 200) {
+      const respJson = await resp.json();
+      if (respJson.success) {
+        const sessionId = respJson.checkoutSessionId;
+        const stripe: any = (window as any).Stripe(process.env.STRIPE_PUBLIC_KEY);
+
+        stripe
+          .redirectToCheckout({
+            sessionId,
+          })
+          .then(function (result: any) {
+            console.error(result.error.message);
+          });
```

Comment:
You have the `console.error` in the `then`. Did you mean `.catch`?

Reply:
Lemme check on this--I wanna say I copied this from the docs; if `redirectToCheckout` succeeds, we won't be on this page anyway. Need to confirm how it does error handling.

Reply:
The [docs](https://stripe.com/docs/js/checkout) seem to suggest that an error will be surface in this manner. Seems non-standard.

Reply:
> if redirectToCheckout succeeds, we won't be on this page anyway

There's some nuance there. A `window.location = …` actually happens asynchronously, so the remainder of the current JS execution will finish before the page unloads.

```js
function foo() {
  console.log('this happens')

  window.location = 'https://example.com'

  console.log('this happens too')

  setTimeout(() => {
    console.log('this never happens')
  }, 0)
}
```

Reply:
Oh huh, TIL!